### PR TITLE
2060 add pdarray data type and size to metrics

### PR DESCRIPTION
**Pull request by hokiegeek2**

### CODE REVIEW

**Review by pierce314159 — APPROVED** [review]
looks good! I have a few small comments but nothing important

File: src/MetricsMsg.chpl [hunk 1]
```diff
@@ -111,30 +154,128 @@ module MetricsMsg {
...
+            
+            var value: real;
+
+            if dataType == MetricDataType.INT {
+                this.intMeasurementTotals(metric) += measurement;
+                value = this.intMeasurementTotals(metric)/numMeasurements;
+            } else {
+                this.realMeasurementTotals(metric) += measurement;
+                value = this.realMeasurementTotals(metric)/numMeasurements;
+            }
```
[the remark this answers is not in the document]
Comment (hokiegeek2) [reply src/MetricsMsg.chpl:266]: @pierce314159 ah, good point, will check

File: src/MetricsMsg.chpl [hunk 2]
```diff
@@ -111,30 +154,128 @@ module MetricsMsg {
...
+                } else {
+                    var value = 0.0;
+                    this.realMeasurements.add(metric, value);
+                    return value;
+                }
             } else {
-                return try! this.measurements.getValue(metric);
+                return this.getMeasurement(metric);
             }
         }
```
[the remark this answers is not in the document]
Comment (hokiegeek2) [reply src/MetricsMsg.chpl:198]: @pierce314159 wow, yup that's a problem, fixing

File: src/MetricsMsg.chpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (hokiegeek2) [reply src/MetricsMsg.chpl:266]: @pierce314159 that's handled just above this code block:

image

[the remark this answers is not in the document]
Comment (pierce314159) [reply src/MetricsMsg.chpl:266]: will this is handling it for `this.numMeasurements`, but not `this.int/realMeasureTotals`, right?

[the remark this answers is not in the document]
Comment (hokiegeek2) [reply src/MetricsMsg.chpl:266]: @pierce314159 ah, yes, I see what you mean. I am gonna add getter methods for both and pull 'em out into separate procs. this is excellent feedback, thank you!

**Review by jaketrookman — APPROVED** [review]
Looks good to me, besides what what previously suggested

**Review by pierce314159 — APPROVED** [review]
looks good!

**Pull request merged**